models/article.go

```diff
@@ -0,0 +1,209 @@
+package models
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Machiel/slugify"
+	"github.com/jinzhu/gorm"
+)
+
+type ScopeHandler func(db *gorm.DB) *gorm.DB
```

Comment:
At the beginning for the articles route I wanted to use scopes for filtering and aggregate the query each time we get a query params (tag, favorited, author) but it didn't work because I was stubborn to only use the DSL. But I think it can be done chaining the Joins() method, I will try.